### ISSUE TITLE
False Positive Trait not implemented Errors

### ISSUE THREAD

**Issue by mihaiandrei1294** [intellij-rust/intellij-rust#5839]
## Environment

* **IntelliJ Rust plugin version:** 0.3.127.3249-202
* **Rust toolchain version:** 1.45.1
* **IDE name and version:** Cliun 2020.2 (Build #CL-202.6397.106, built on July 28, 2020)
* **Operating system:** Windows 10 10.0 (1909 build 18363.959)

## Problem description

It seems that Intellij-Rust cannot properly infer the trait bounds of some objects and give false "trait bound not satisfied" type of errors. In the 2 images below you can see the 2 errors given but the code runs with no problem.

![No_copy_tait](https://user-images.githubusercontent.com/14346004/89009417-2bc00580-d315-11ea-9a1d-3b96a04d4f1d.png)

![No_debug_tait](https://user-images.githubusercontent.com/14346004/89009418-2c589c00-d315-11ea-9600-415b7762a038.png)

It seems that this problem appears when the associated type of a trait has a trait bound and *that* trait bound also has some trait bounds, which are the ones required. It appears that intellij-rust cannot infer the "second-level" traits so to speak, therefore giving false "trait bound not satisfied" errors.

## Steps to reproduce

```rust
use std::fmt::Debug;

trait IsCopyAndDebug:Copy + Debug{}

trait HasTypeWithBounds{
    type TypeWithBounds: IsCopyAndDebug;

    fn get_copy_type(&self) -> &Self::TypeWithBounds;
}

#[derive(Copy, Clone, Debug)]
struct Small{}

impl IsCopyAndDebug for Small {}

struct Main{
    s: Small,
}

impl HasTypeWithBounds for Main{
    type TypeWithBounds = Small;

    fn get_copy_type(&self) -> &Self::TypeWithBounds {
        &self.s
    }
}


fn generic_fn<T: HasTypeWithBounds>(o: T){
    let sref = o.get_copy_type();

    let s2 = *sref;

    println!("s: {:?}, s2: {:?}",sref, s2);
}


fn main() {

    let s = Small{};

    let o = Main{s};

    generic_fn(o);

}

```

**Comment by Kobzol** [comment]
Works in the latest version of the plugin.
